sheetshuttle/util.py

```diff
@@ -26,6 +26,19 @@ def gh_token_exists() -> bool:
     return False
 
 
+def extract_sheet_id(url: str) -> str:
+    """Extract a sheet ID from a Google Sheets URL."""
+    url_segments = url.split("/")
+    if "d" not in url_segments:
+        raise ValueError(f"The URL {url} is not valid for Google Sheets ID extraction.")
+    # Return the url_segment following the segment that contains only "d"
+    idx = url_segments.index("d")
+    if idx == (len(url_segments) - 1):
+        raise IndexError(f"The URL {url} is not valid for Google Sheets ID extraction.")
```

Comment:
Let's not use `ValueError` and `IndexError` exceptions here and just use the generic one aka `Exception`. The two you're using are meant. A value error is thrown when "a function receives an argument of the correct data type but an inappropriate value" and index error is thrown when "an attempt is made to access an item in a list at an index which is out of bounds". Neither of these cases apply here and we should avoid confusing the user.

Reply:
I've run into issues with the generic `Exception` failing lint checks in the past--hence the attempt to use the more explicitly defined Exceptions. I would also argue that the `ValueError` is appropriate: we're expecting functions to contain the `"/d/"` token--the `if` statement that precedes this is checking that the url passed is "of an appropriate value", so to speak (or I suppose more accurately *contains* an appropriate value). 

Additionally, the raise for the `IndexError` is coming after an if statement checks whether or not the index (or `idx`) of the `"d"` statement is *going* to cause an `IndexError` (since immediately after the if statement we're grabbing the index of the value *after* the `"d"` statement).

From my perspective this feels like the correct use of either exception (to stay in accordance with the standards being reinforced by the linters used); could you clarify why one or the other is inappropriately used?

Reply:
I understand the argument for using these exceptions, their names match up with the issue the user would run into so it seems ok to do that. However, I still think we should not use them since they're meant for **very specific** cases which I mentioned and nothing else.

I don't think throwing a generic exception would cause a linter error. I've only had that issue when trying to catch a generic exception.

With all of this in mind, I propose this solution as a middle ground:
We can create our own custom exception that inherits from `Exception`! Let's maybe call it `InvalidSheetInfo` or something similar and descriptive. We might need to use it later on as we continue to develop SheetShuttle. Then you can import the module that defines this exception and throw it when needed with the appropriate message. We currently do that [here](https://github.com/GatorEducator/SheetShuttle/blob/main/sheetshuttle/sheet_collector.py#L108) with `MissingAuthenticationVariable` which we throw [here]( MissingAuthenticationVariable).
Then when you add the testing, you can just assert that this type of exception is being thrown.
What do you think of this approach?